library/src/meta/reference.rs

```diff
@@ -84,12 +84,64 @@ pub struct RefElem {
     /// A synthesized citation.
     #[synthesized]
     pub citation: Option<CiteElem>,
+
+    /// Content of the referee, it should be referable.
+    ///
+    /// ```example
+    /// #set heading(numbering: (..nums) => {
+    ///   nums.pos().map(str).join(".")
+    ///   }, supplement: [Chapt])
+    ///
+    /// #show ref: it => {
+    ///   if it.has("referee") and it.referee.func() == heading {
+    ///     let referee = it.referee
+    ///     "["
+    ///     referee.supplement
+    ///     "-"
+    ///     numbering(referee.numbering, ..counter(heading).at(referee.location()))
+    ///     "]"
+    ///   } else {
+    ///     it
+    ///   }
+    /// }
+    ///
+    /// = Introduction <intro>
+    /// = Summary <sum>
+    /// == Subsection <sub>
+    /// @intro
+    ///
+    /// @sum
+    ///
+    /// @sub
+    /// ```
+    #[synthesized]
+    pub referee: Option<Content>,
 }
 
 impl Synthesize for RefElem {
     fn synthesize(&mut self, vt: &mut Vt, styles: StyleChain) -> SourceResult<()> {
         let citation = self.to_citation(vt, styles)?;
         self.push_citation(Some(citation));
+
+        if !vt.introspector.init() {
+            return Ok(());
+        }
+
+        // find the referee element
+        let target = self.target();
+        let elem = vt.introspector.query_label(&self.target());
+
+        if BibliographyElem::has(vt, &target.0) {
+            // only push bib element if it is not in the document
+            if elem.is_err() {
+                self.push_referee(Some(self.to_citation(vt, styles)?.pack()));
```

Comment:
not sure whether it makes sense to push a citation here? should we just set referee to `None` here, too? @Dherse what do you think?

Reply:
The thing is that, from what I can see, when it's a reference to a citation it sort of "turns" the reference into a citation, so my question is this: does it even make sense to `show` a `ref` that is essentially a `cite`? Or should we just straight up do the conversion to a `cite` in a pre-processing stage? I think removing the ability for `ref` to sort-of be a `cite` would be a good change.

We could try and disambiguate earlier and select the right element based on what is being referenced.

Reply:
in `show` method of RefElem, if a label is in Bibliography, it will return a citation, for which I preserve this behavior in my implementation of synthesize but now I think it just makes no sense here since `ref` has a citation as its attribute

Reply:
Fair enough.